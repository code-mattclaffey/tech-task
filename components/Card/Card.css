/* Card Component Styles */

:root {
    --card-bg: var(--white);
    --card-border-raidius: 5px;
    --card-padding: 20px;
}

@media screen and (min-width: 40em) {
    :root {
        --card-padding: 30px;
    }
}

.c-card {
    background: var(--card-bg);
    border-radius: var(--card-border-raidius);
    color: var(--primary);
    padding: var(--card-padding);
}
