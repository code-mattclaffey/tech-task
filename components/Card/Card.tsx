import React from 'react';
import classNames from 'classnames';

// @ts-ignore
import { Heading, Text } from "@titan-tooling/ui";

type Props = {
    as?: string;
    additionalClassNames?: string;
};

export const Card: React.FunctionComponent<Props> = ({ additionalClassNames, children }) => {
    const classes = classNames('c-card', additionalClassNames);

    return (
        <div className={classes}>
            {children}
        </div>
    );
};
