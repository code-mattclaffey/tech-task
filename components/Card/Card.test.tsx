import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Card } from './Card';

afterEach(() => {
    cleanup();
});

describe('<Card />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <Card>
                This is some jsx content
            </Card>
        );

        expect(container).toMatchSnapshot();
    });
});
