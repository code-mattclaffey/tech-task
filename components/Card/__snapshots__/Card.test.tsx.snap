// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<div>
  <div
    class="c-card"
  >
    This is some jsx content
  </div>
</div>
`;
